index.html

```diff
@@ -975,6 +976,67 @@ <h2>Methods</h2>
             selected candidate pair =].
           </p>
         </dd>
+        <dt>
+          <dfn>removeCandidatePairs</dfn>
+        </dt>
+        <dd>
+          <p>
+            The {{removeCandidatePairs}} method immediately removes the provided candidate pairs, making them unavailable to use
+            for transport. The [= ICE agent =] may free up candidates once all the associated candidate pairs have been removed.
+          </p>
+          <p>
+            When this method is invoked, the [= user agent =] MUST run the following steps:
+          </p>
+          <ol class="algorithm">
+            <li>
+              <p>
+                Let |connection:RTCPeerConnection| be the {{RTCPeerConnection}} object associated with this [= ICE agent =].
+              </p>
+            </li>
+            <li>
+              <p>
+                If <var>connection</var>.{{RTCPeerConnection/[[IsClosed]]}} is
+                <code>true</code>, [= exception/throw =] an
+                {{InvalidStateError}}.
+              </p>
+            </li>
+            <li>
+              <p>
+                Let |transport:RTCIceTransport| be the associated {{RTCIceTransport}} object.
+              </p>
+            </li>
+            <li>
+              <p>
+                If |transport|.{{RTCIceTransport/[[IceTransportState]]}} is either of
+                {{RTCIceTransportState/"new"}}, {{RTCIceTransportState/"failed"}} or {{RTCIceTransportState/"closed"}}, [=
+                exception/throw =]
+                an {{InvalidStateError}}.
+              </p>
+            </li>
+            <li>
+              <p>
+                If |transport|.{{RTCIceTransport/[[ProposalPending]]}} is <code>true</code>, [= exception/throw =] an
+                {{InvalidStateError}}.
+              </p>
+            </li>
+            <li>
+              <p>
+                Instruct the [= ICE agent =] to remove the candidate pairs indicated by |candidatePairs|, ignoring any unknown
+                or invalid candidate pairs.
+              </p>
+            </li>
+            <li>
+              <p>
+                For each valid |removedPair| in |candidatePairs| that was removed by the [= ICE agent =], [= fire an event =] named
```

Comment:
It sounds like to know which pairs was removed here, we would need to block JS on ICE agent work.

Should this be in a separate task, i.e. "When the ICE agent as removed all candidate pairs, queue a task that runs the following steps: for each |removePair| in |removedCandidatePairs|, fire ..."

Reply:
Right, we shouldn't block JS on the ICE agent's work. Moved the events firing to a separate task.

Also changed removeCandidatePairs to return a promise that resolves after all changes occurring due to the method have taken effect. This follows a similar pattern for batched actions elsewhere, eg. setRemoteDescription, i.e. return a promise, do everything that has an effect, fire all events, finally resolve the promise.

Reply:
Excellent!